webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -2480,19 +2480,88 @@ var CURRENT_GROUP_DATA={};
                     _this.toggleMediaOptions(this);
 
                 });
-                $mck_text_box.keyup(function (){
+             
+            _this.cursorPosition =  function () {
+                var sel = document.getSelection();
+                sel.modify("extend", "backward", "documentboundary");
+                var pos = sel.toString().length;
+                if(sel.anchorNode != undefined) sel.collapseToEnd();
+                return pos;
+            }
+            _this.disableSendButton = function () {
+                let sendButton = document.getElementById('mck-msg-sbmt');
+                sendButton.setAttribute("disabled", true);
+                CURRENT_GROUP_DATA.DISABLE_SEND_MESSAGE = true;
+            }
+            _this.enableSendButton = function () {
+                let sendButton = document.getElementById('mck-msg-sbmt');
+                sendButton.removeAttribute("disabled", false);
+                CURRENT_GROUP_DATA.DISABLE_SEND_MESSAGE = false;
+            }
+                $mck_text_box.on('input paste', function (event){
                     if (CURRENT_GROUP_DATA.CHAR_CHECK) {
+                        let warningLength = 199;
+                        let maxLength = 256;
+                        let textBox = $mck_text_box[0]; //using seperate selector for vanilla JS functions
                         if (!(document.getElementById('mck-char-count'))) {
                             warningText.innerHTML += '<span> | </span><span id="mck-char-count"></span>';
                         }
                         var remtxt;
-                        var str = ($mck_text_box.text().toString()).trim();
-                        var len = str.length;
-                        if (len > 199) {
-                            if (len > 256) {
-                                remtxt = "(" + MCK_LABELS['limit.remove'] + " " + (len - 256) + " " + MCK_LABELS['limit.characters'] + ")";
+                        let textVal = mckUtils.textVal(textBox);
+                        let str = textVal.trim();
+                        var textLength = str.length;
+                        if (textLength > warningLength) {
+                            let caretPosition = _this.cursorPosition();
```

Comment:
@hardikparashar for web plugin, please use 'var'. let might not work in IE.

Reply:
@hardikparashar there are more places where let is used. please remove from all places.